packages/flame/test/components/composed_component_test.dart

```diff
@@ -181,5 +184,26 @@ void main() {
 
       expect(child.debugMode, true);
     });
+    test('fail to add child if no gameRef can be acquired', () {
+      final game = MyGame();
+      game.onResize(Vector2.all(100));
+
+      final parent = PlainComposed();
+
+      // this is ok; when the parent is added to the game the children will
+      // get mounted
+      parent.addChild(MyTap());
+
+      game.add(parent);
+      game.update(0);
```

Comment:
so, technically `isMounted` is only triggered after the component is added to the list, not after it's prepared. currently it's impossible to know if the component has been prepared, that I know of. this left me thinking if we allow a component to be added to multiple games, or to the same game twice. currently there is no check for that either. for this and (possibly, if we want) that, there are 3 options

1. we can use isMounted as is because this whole thing is already an edge case of incorrect behavior, and the game not ticking is the edge case of the edge case
2. we could create a new lifecycle state, isPrepared
3. we could redefine isMounted to mean is prepared

Reply:
I vote for 3, don't see a reason to have both those lifecycle states

Reply:
I can do that & add a double add restriction on a followup

Reply:
there might be use cases where you might wanna know only after the component is added though. so that is why I was not sure. either way I think this PR is already a vast improvement, but maybe we can reconsider things as a followup